plugins/amazonq/codewhisperer/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codewhisperer/telemetry/CodeWhispererCodeCoverageTracker.kt

```diff
@@ -85,25 +89,52 @@ abstract class CodeWhispererCodeCoverageTracker(
-            CodeWhispererPopupManager.CODEWHISPERER_USER_ACTION_PERFORMED,
-            object : CodeWhispererUserActionListener {
-                override fun afterAccept(states: InvocationContext, sessionContext: SessionContext, rangeMarker: RangeMarker) {
-                    if (states.requestContext.fileContextInfo.programmingLanguage != language) return
-                    rangeMarkers.add(rangeMarker)
-                    val originalRecommendation = extractRangeMarkerString(rangeMarker) ?: return
-                    rangeMarker.putUserData(KEY_REMAINING_RECOMMENDATION, originalRecommendation)
-                    runReadAction {
-                        // also increment total tokens because accepted tokens are part of it
-                        incrementTotalCharsCount(rangeMarker.document, originalRecommendation.length)
-                        // avoid counting CodeWhisperer inserted suggestion twice in total tokens
-                        if (rangeMarker.textRange.length in 2..49 && originalRecommendation.trim().isNotEmpty()) {
-                            incrementTotalCharsCount(rangeMarker.document, -rangeMarker.textRange.length)
+        if (CodeWhispererFeatureConfigService.getInstance().getNewAutoTriggerUX()) {
+            conn.subscribe(
+                CodeWhispererPopupManager.CODEWHISPERER_USER_ACTION_PERFORMED,
+                object : CodeWhispererUserActionListener {
+                    override fun afterAccept(
+                        states: InvocationContextNew,
+                        previews: List<PreviewContext>,
+                        sessionContext: SessionContextNew,
+                        rangeMarker: RangeMarker,
+                    ) {
+                        if (states.requestContext.fileContextInfo.programmingLanguage != language) return
+                        rangeMarkers.add(rangeMarker)
+                        val originalRecommendation = extractRangeMarkerString(rangeMarker) ?: return
+                        rangeMarker.putUserData(KEY_REMAINING_RECOMMENDATION, originalRecommendation)
+                        runReadAction {
+                            // also increment total tokens because accepted tokens are part of it
+                            incrementTotalCharsCount(rangeMarker.document, originalRecommendation.length)
+                            // avoid counting CodeWhisperer inserted suggestion twice in total tokens
+                            if (rangeMarker.textRange.length in 2..49 && originalRecommendation.trim().isNotEmpty()) {
+                                incrementTotalCharsCount(rangeMarker.document, -rangeMarker.textRange.length)
+                            }
                         }
                     }
                 }
-            }
-        )
+            )
+        } else {
+            conn.subscribe(
+                CodeWhispererPopupManager.CODEWHISPERER_USER_ACTION_PERFORMED,
+                object : CodeWhispererUserActionListener {
```

Comment:
looks like these 2 lines are shared, so we just need one listener with 2 afterAccept.